src/jestMatchers.ts

```diff
@@ -0,0 +1,338 @@
+        interface Matchers<R = void, T = {}> extends AwsSdkJestMockMatchers<T, R> { }
+    }
+}
+
+type ClientMock = AwsStub<any, any>;
+type AnyCommand = AwsCommand<any, any>;
+type AnySpyCall = sinon.SinonSpyCall<[AnyCommand]>;
+type MessageFunctionParams<CheckData> = {
+    cmd: string;
+    client: string;
+    calls: AnySpyCall[];
+    commandCalls: AnySpyCall[];
+    data: CheckData;
+};
+
+/**
+ * Prettyprints command calls for message
+ *
+ * @param ctx
+ * @param calls
+ * @returns
+ */
+function printCalls(ctx: jest.MatcherContext, calls: AnySpyCall[]): string[] {
+    return calls.map(
+        (c, i) =>
+            `  ${i + 1}, ${c.args[0].constructor.name}, ${ctx.utils.printReceived(
+                c.args[0].input
+            )}`
+    );
+}
+
+export function processMatch<CheckData>({
+    ctx,
+    mockClient,
+    command,
+    check,
+    input,
+    message,
+    notMessage,
+}: {
+    ctx: jest.MatcherContext;
+    mockClient: ClientMock;
+    command: new () => AnyCommand;
+    check: (params: { calls: AnySpyCall[]; commandCalls: AnySpyCall[] }) => {
+        pass: boolean;
+        data: CheckData;
+    };
+    input: Record<string, unknown> | undefined;
+    message: (params: MessageFunctionParams<CheckData>) => string[];
+    notMessage: (params: MessageFunctionParams<CheckData>) => string[];
+}): jest.CustomMatcherResult {
+    assert(
+        command &&
+        typeof command === 'function' &&
+        typeof command.name === 'string' &&
+        command.name.length > 0,
+        'Command must be valid AWS Sdk Command'
+    );
+
+    const calls = mockClient.calls();
+    const commandCalls = mockClient.commandCalls(command, input);
+    const { pass, data } = check({ calls, commandCalls });
+
+    const msg = (): string => {
+        const cmd = ctx.utils.printExpected(command.name);
+        const client = mockClient.clientName();
+
+        const msgParams: MessageFunctionParams<CheckData> = {
+            calls,
+            client,
+            cmd,
+            data,
+            commandCalls,
+        };
+
+        return (ctx.isNot ? notMessage(msgParams) : message(msgParams)).join('\n');
+    };
+
+    return { pass, message: msg };
+}
+
+/** Using them for testing */
+export const matchers: jest.ExpectExtendMap = {
+    /**
+     * implementation of {@link AwsSdkJestMockMatchers.toHaveReceivedCommandTimes} matcher
+     */
+    toHaveReceivedCommandTimes(
+        this: jest.MatcherContext,
+        mockClient: ClientMock,
+        command: new () => AnyCommand,
+        expectedCalls: number
+    ) {
+        return processMatch({
+            ctx: this,
+            mockClient,
+            command,
+            input: undefined,
+            check: ({ commandCalls }) => ({ pass: commandCalls.length === expectedCalls, data: {} }),
+            message: ({ client, cmd, commandCalls }) => [
+                `Expected ${client} have received ${cmd} ${this.utils.printExpected(
```

Comment:
We could add a variable for `not` based on the `ctx.isNot` and conditionally put it in the message:

```ts
const not = isNot ? 'not ' : '';
`Expected ${client} to ${not}receive ${cmd}...`
```

This way we can get rid of `notMessage` parameter.